### PR TITLE
SW6 IsInvoiceEmailEnabled

### PR DESCRIPTION
**Pull request by cif-ps**

### CODE REVIEW

File: backend/de.metas.externalsystem/src/main/java/de/metas/externalsystem/ExternalSystemConfigRepo.java [hunk 1]
```diff
@@ -255,6 +255,7 @@ private ExternalSystemShopware6ConfigMapping toExternalSystemShopware6ConfigMapp
 				.sw6PaymentMethod(record.getSW6_Payment_Method())
 				.description(record.getDescription())
 				.seqNo(record.getSeqNo())
+				.isInvoiceEmailEnabled(record.isInvoiceEmailEnabled())
```
Comment (metas-ts) [inline backend/de.metas.externalsystem/src/main/java/de/metas/externalsystem/ExternalSystemConfigRepo.java:258]: support null

File: backend/de.metas.externalsystem/src/main/java/de/metas/externalsystem/shopware6/ExternalSystemShopware6ConfigMapping.java [hunk 2]
```diff
@@ -76,6 +78,7 @@ public ExternalSystemShopware6ConfigMapping(
 			@NonNull final String bpartnerIfNotExists,
 			@NonNull final String bpartnerLocationIfExists,
 			@NonNull final String bpartnerLocationIfNotExists,
+			final boolean isInvoiceEmailEnabled,
```
Comment (metas-ts) [inline backend/de.metas.externalsystem/src/main/java/de/metas/externalsystem/shopware6/ExternalSystemShopware6ConfigMapping.java:81]:
```suggestion
			@Nullable final Boolean isInvoiceEmailEnabled,
```

**Review by metas-ts — CHANGES_REQUESTED** [review]
TLDR: basically, the new flag needs to be optional (i.e. nullable)

File: backend/de.metas.business/src/main/java/de/metas/bpartner/composite/repository/BPartnerCompositesLoader.java [hunk 3]
```diff
@@ -332,6 +332,14 @@ private static BPartnerContact ofContactRecord(
 	{
 		final RecordChangeLog changeLog = ChangeLogUtil.createContactChangeLog(contactRecord, relatedRecords);
 
+		Boolean invoiceEmailEnabled = null;
+
+		if(contactRecord.getIsInvoiceEmailEnabled() != null)
+		{
+			invoiceEmailEnabled = ISINVOICEEMAILENABLED_Yes.equals(contactRecord.getIsInvoiceEmailEnabled());
+		}
```
Comment (metas-ts) [inline backend/de.metas.business/src/main/java/de/metas/bpartner/composite/repository/BPartnerCompositesLoader.java:340]: please use `de.metas.common.util.StringUtils#toBoolean(java.lang.Object, java.lang.Boolean)`
instead, with `defaultValue = null`

File: backend/de.metas.externalsystem/src/main/java/de/metas/externalsystem/ExternalSystemConfigRepo.java [hunk 4]
```diff
@@ -247,6 +249,14 @@ private ExternalSystemShopware6Config buildExternalSystemShopware6Config(final @
 	@NonNull
 	private ExternalSystemShopware6ConfigMapping toExternalSystemShopware6ConfigMapping(@NonNull final I_ExternalSystem_Config_Shopware6Mapping record)
 	{
+
+		Boolean invoiceEmailEnabled = null;
```
Comment (metas-ts) [inline backend/de.metas.externalsystem/src/main/java/de/metas/externalsystem/ExternalSystemConfigRepo.java:253]: please use `de.metas.common.util.StringUtils#toBoolean(java.lang.Object, java.lang.Boolean)`
instead, with `defaultValue = null`

**Review by metas-ts — COMMENTED** [review]
allmost there, please check my remarks

**Review by metas-ts — APPROVED** [review]
